app/Products.tsx

```diff
@@ -1,19 +1,24 @@
-import { useRouter } from 'next/router';
+'use client';
+import { useRouter } from 'next/navigation';
 import { useState } from 'react';
 import Stripe from 'stripe';
-import { Counter } from '../components/Counter';
-import { EndpointResponse, StripeSession } from './api/sessions';
+import { EndpointResponse, StripeSession } from '../pages/api/sessions';
+import { Counter } from './Counter';
 
 type Product = {
   price: Stripe.Price;
   product: Stripe.Product;
 };
 
 type Props = {
-  tablet: Product;
-  magazine: Product;
+  tablet: string;
+  magazine: string;
 };
-export default function Home(props: Props) {
+
+export default function Products(props: Props) {
+  const tablet: Product = JSON.parse(props.tablet);
```

Comment:
can we get away without the `JSON.stringify` / `JSON.parse`?

Reply:
https://github.com/upleveled/connecting-to-external-apis-secrets-with-stripe-fall-2022-vienna-austria/blob/85f6653d67eeb9ccaf099006071fdc8d8db4124b/app/Products.tsx#L10-L19